### PR TITLE
Add TeachingEventsByType response model

### PR DESCRIPTION
**Pull request by ethax-ross**
Introduces a new response model for returning teaching events grouped by type. Previously we were using the format:

``` { "<typeid>": [<events>] } ```

This isn't a very RESTful response and also causes the type id to be returned in a string, where in actuality its an integer value.

This commit updates the response to be:

``` [{ "typeId": <typeid>, "teachingEvents": [<events>] }, { ... }] ```

In order to introduce the new response type an additional endpoint have been added:

``` 
teaching_events/search_grouped_by_type
```

These replace the existing methods, that will be removed once the clients have been updated (note the `upcoming_indexed_by_type` endpoint is going to be deprecated, which is why there is no equivalent `_grouped_` endpoint above for it):

``` 
teaching_events/search_indexed_by_type
teaching_events/upcoming_indexed_by_type
```

### CODE REVIEW

File: GetIntoTeachingApiTests/Models/TeachingEventsByTypeTests.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿namespace GetIntoTeachingApiTests.Models
+{
+    public class TeachingEventsByTypeTests
+    {
```
Comment (jebw) [inline GetIntoTeachingApiTests/Models/TeachingEventsByTypeTests.cs:4]: Should this spec be empty?

**Review by jebw — COMMENTED** [review]
Should the TeachingEventsByTypeTests be empty?

File: GetIntoTeachingApiTests/Models/TeachingEventsByTypeTests.cs (hunk 1 above)
Comment (ethax-ross) [reply GetIntoTeachingApiTests/Models/TeachingEventsByTypeTests.cs:4]: Yeah the model only has getters/setters that are tested indirectly, no real value testing them explicitly but I like to have the test file there for if/when its needed.

**Review by jebw — APPROVED** [review]
LGTM

**Pull request merged**